lib/spack/spack/test/directory_layout.py

```diff
@@ -120,7 +120,7 @@ def test_read_and_write_spec(temporary_store, config, mock_packages):
         # TODO: fix this when we can concretize more loosely based on
         # TODO: what is installed. We currently omit these to
         # TODO: increase reuse of build dependencies.
-        stored_deptypes = ('link', 'run')
+        stored_deptypes = spack.hash_types.full_hash
```

Comment:
@tgamblin @haampie @alalazo this is a type error. It needs to be `spack.dependency.all_deptypes`.

(The code works because if we pass the `Spec.copy` method something invalid, it defaults to `spack.dependency.all_deptypes` (after a pass-through to `Spec._dup`))